### DISCUSSION TITLE
Creating nested object from single query?

### DISCUSSION THREAD

**Discussion by canavci312** [simolus3/drift#2823] [Q&A] [answered]
I know this is too simple but after dealing with nosql for so long I couldnt figure out how to create model from 2 tables such. How can I query to get the object `List<CarGallery>?`
```
class CarGallery {
  final String name;
  final List<Models> models;
  CarGallery(this.name, this.models);
}

class Models {
  final name;
  final age;
  Models(this.name, this.age);
}
class CarGalleryTable extends Table {
  TextColumn get id => text()();
  TextColumn get name => text()();
  @override
  Set<Column> get primaryKey => {id};
}
class ModelsTable extends Table {
  TextColumn get id => text()();
  TextColumn get name => text()();
  TextColumn get carGalleryId => text().references(CarGalleryTable, #id)();
  @override
  Set<Column> get primaryKey => {id};
}
```

**Comment by simolus3** [comment, marked as answer]
Drift has no builtin way to resolve that relationship. This is by design, drift doesn't hide the relational structure of the database from you. [This page](https://drift.simonbinder.eu/docs/examples/relationships/) has more details, basically you need two queries:

1. A query to load the car gallery you're interested in.
2. A query to load all models with that gallery id.

Afterwards you can stitch them together in Dart to get the `CarGallery` instance.

An alternative would be to store models in the `CarGalleryTable` directly by using a [type converter](https://drift.simonbinder.eu/docs/advanced-features/type_converters/) storing a list of `Models` as JSON. This may be more familiar when coming from a nosql database (and even though it goes against relational best practices, IMO there is nothing really wrong with that for 1:n relationships).

**Reply by canavci312** [reply]
Thank you for your time and answer, I got better understanding now.
